src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -157,22 +196,32 @@ public int getNumberOfAllLabels() {
    */
   @Restricted(NoExternalUse.class)
   public int getAssignedResourceAmount(String label) {
-    return LockableResourcesManager.get().getResourcesWithLabel(label, null).size();
+    synchronized(lrm) {
+      return lrm.getResourcesWithLabel(label, null).size();
+    }
   }
 
   /** Returns current queue */
   @Restricted(NoExternalUse.class) // used by jelly
   public List<QueuedContextStruct> getCurrentQueuedContext() {
-    return LockableResourcesManager.get().getCurrentQueuedContext();
+    synchronized(lrm) {
+      return new ArrayList<>(lrm.getCurrentQueuedContext());
+    }
   }
 
   /** Returns current queue */
   @Restricted(NoExternalUse.class) // used by jelly
   @CheckForNull
   public LockableResourcesStruct getOldestQueue() {
```

Comment:
This is still not thread safe, It shall be refacotred, and make more simple, because it takes too long to show all this info in jelly